test/test_cuda_mapper.cc

```diff
@@ -753,6 +753,38 @@ def perforatedConvolution(float(N, C, H, W) input, float(M, C, KH, KW) weights,
   Prepare(tc);
 }
 
+TEST_F(PolyhedralMapperTest, ModulusConstantRHS) {
+  string tc = R"TC(
+def fun(float(N) a) -> (b) { b(i) = a(i % 3) where i in 0:N }
+)TC";
+  // This triggers tc2halide conversion and should not throw.
+  auto scop = Prepare(tc);
+  for (auto r : scop->reads.range().get_set_list()) {
```

Comment:
thanks @ftynse . can you provide guidance on skimo's comment ? I am not sure myself which is the right approach. Thanks a lot :)

Reply:
It's a matter of preference, I think.  @skimo-openhub suggests to construct a new relation from its textual description, basically `isl::map(ctx, "<paste here the textual representation of the map litearlly>")` (taking `ctx` from whatever isl object available around) and comparing that to the result.  It would work, but I prefer testing only the relevant properties of the relation, hence the stride check.  Checking only relevant properties removes the need to modify the test when irrelevant properties change.  I will accept both ways of checking.

More specifically, current access relations do not involve bounds imposed by tensor sizes.  I'm quite annoyed at this during memory promotion and was planning to add those bounds in access relation construction.  That would change these relations, requiring to update this test that is affected only indirectly.